drivers/modem/wncm14a2a.c

```diff
@@ -0,0 +1,1905 @@
+
+	/* new socket */
+	sock = socket_get();
+	if (!sock) {
+		return -ENOMEM;
+	}
+
+	(*context)->offload_context = sock;
+	sock->family = family;
+	sock->type = type;
+	sock->ip_proto = ip_proto;
+	sock->context = *context;
+	sock->socket_id = MDM_MAX_SOCKETS + 1; /* socket # needs assigning */
+
+	snprintk(buf, sizeof(buf), "AT@SOCKCREAT=%d,%d", type,
+		 family == AF_INET ? 0 : 1);
+	ret = send_at_cmd(NULL, buf, MDM_CMD_TIMEOUT);
+	if (ret < 0) {
+		SYS_LOG_ERR("AT@SOCKCREAT ret:%d", ret);
+		socket_put(sock);
+	}
+
+	return ret;
+}
+
+static int offload_bind(struct net_context *context,
+			const struct sockaddr *addr,
+			socklen_t addrlen)
+{
+	struct wncm14a2a_socket *sock = NULL;
+
+	if (!context) {
+		return -EINVAL;
+	}
+
+	sock = (struct wncm14a2a_socket *)context->offload_context;
+	if (!sock) {
+		SYS_LOG_ERR("Can't locate socket for net_ctx:%p!", context);
+		return -EINVAL;
+	}
+
+	/* save bind address information */
+	sock->src.sa_family = addr->sa_family;
+#if defined(CONFIG_NET_IPV6)
+	if (addr->sa_family == AF_INET6) {
+		net_ipaddr_copy(&net_sin6(&sock->src)->sin6_addr,
+				&net_sin6(addr)->sin6_addr);
+		net_sin6(&sock->src)->sin6_port = net_sin6(addr)->sin6_port;
+	}
+#endif
+#if defined(CONFIG_NET_IPV4)
+	if (addr->sa_family == AF_INET) {
+		net_ipaddr_copy(&net_sin(&sock->src)->sin_addr,
+				&net_sin(addr)->sin_addr);
+		net_sin(&sock->src)->sin_port = net_sin(addr)->sin_port;
+	}
+#endif
+
+	return 0;
+}
+
+static int offload_listen(struct net_context *context, int backlog)
+{
+	/* NOT IMPLEMENTED */
+	return -EINVAL;
+}
+
+static int offload_connect(struct net_context *context,
+			   const struct sockaddr *addr,
+			   socklen_t addrlen,
+			   net_context_connect_cb_t cb,
+			   s32_t timeout,
+			   void *user_data)
+{
+	int ret;
+	s32_t timeout_sec = timeout / MSEC_PER_SEC;
+	char buf[100]; /* AT@SOCKCONN=#,###.###.###.###,#####,##### */
+	struct wncm14a2a_socket *sock;
+
+	if (!context || !addr) {
+		return -EINVAL;
+	}
+
+	sock = (struct wncm14a2a_socket *)context->offload_context;
+	if (!sock) {
+		SYS_LOG_ERR("Can't locate socket for net_ctx:%p!", context);
+		return -EINVAL;
+	}
+
+	if (sock->socket_id < 1) {
+		SYS_LOG_ERR("Invalid socket_id(%d) for net_ctx:%p!",
+			    sock->socket_id, context);
+		return -EINVAL;
+	}
+
+	sock->dst_port = -1;
+	sock->dst.sa_family = addr->sa_family;
+
+#if defined(CONFIG_NET_IPV6)
+	if (addr->sa_family == AF_INET6) {
+		net_ipaddr_copy(&net_sin6(&sock->dst)->sin6_addr,
+				&net_sin6(addr)->sin6_addr);
+		net_sin6(&sock->dst)->sin6_port = net_sin6(addr)->sin6_port;
+		sock->dst_port = ntohs(net_sin6(addr)->sin6_port);
+	}
+#endif
+#if defined(CONFIG_NET_IPV4)
+	if (addr->sa_family == AF_INET) {
+		sock->dst.sa_family = addr->sa_family;
+		net_ipaddr_copy(&net_sin(&sock->dst)->sin_addr,
+				&net_sin(addr)->sin_addr);
+		net_sin(&sock->dst)->sin_port = net_sin(addr)->sin_port;
+		sock->dst_port = ntohs(net_sin(addr)->sin_port);
+	}
+#endif
+
```

Comment:
Could you add else branch here and return -EINVAL if family is not properly set.